ZividOpenCV/ZDF2OpenCV/ZDF2OpenCV.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+Import a ZDF point cloud and convert it to OpenCV format.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <iostream>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/opencv.hpp>
+
+enum class Axis
+{
+    X,
+    Y,
+    Z
+};
+
+template<Axis c>
+static float getValue(const Zivid::Point &p);
+
+template<Axis c>
+static bool minNan(const Zivid::Point a, const Zivid::Point b);
+
+template<>
+static float getValue<Axis::X>(const Zivid::Point &p)
+{
+    return p.x;
+}
+
+template<>
+static float getValue<Axis::Y>(const Zivid::Point &p)
+{
+    return p.y;
+}
+
+template<>
+static float getValue<Axis::Z>(const Zivid::Point &p)
+{
+    return p.z;
+}
+template<Axis c>
+static bool maxNan(Zivid::Point a, Zivid::Point b)
+{
+    return getValue<c>(a) < getValue<c>(b) ? true : std::isnan(getValue<c>(a));
```

Comment:
Should I then use isnan instead of std:isnan ?

Reply:
How come both std:isnan and isnan work and std::max_element works but max_element doesn't?

Reply:
#include <math.h> works not #include <math> - that is ok right?

Reply:
The old C headers are called `math.h` etc.. You can use them in C++ since C is (mostly) a subset of C++. However, the proper C++ versions are called `cmath` etc., so you should prefer those.

The proper C++ ones put the functions inside the `std::` namespace, so you should use `std::isnan`, not `isnan`.

My comment wasn't about using `std::isnan` vs. just `isnan`, it was about that you should always include the headers that the functions you use are defined in. The only reason your code compiled now is that some other header you've included happens to include `cmath` on the platforms you've tested the code on. (It's often hard to remember this, since you don't get any warnings. And more often than not, it doesn't really matter in practice, at least if you have cross platform CI. I often forget myself, but sometimes we catch it in review.)

Reply:
Got it. I don't even know what comes from where and I should probably.
Done, both cmath and std::isnan.